ang/civicase/Actions.js

```diff
@@ -1,386 +1,364 @@
-                  '<div><input name="change_case_status" placeholder="' + ts('Select New Status') + '" /></div>' +
-                  '<label for="change_case_status_details">' + ts('Notes') + '</label>' +
-                  '<textarea id="change_case_status_details"></textarea>' +
-                  '</form>';
-              var statuses = _.map(CRM.civicase.caseStatuses, function (item, statusId) {
-                return { id: item.name, text: item.label, disabled: statusId === currentStatus };
-              });
+            if (!angular.equals(cases[0].client, cases[1].client)) {
+              msg += '<br />' + ts('Warning: selected cases belong to different clients.');
+            }
 
-              _.each(types, function (caseTypeId) {
-                var allowedStatuses = CRM.civicase.caseTypes[caseTypeId].definition.statuses || [];
-                if (allowedStatuses.length) {
-                  _.remove(statuses, function (status) {
-                    return allowedStatuses.indexOf(status.id) < 0;
-                  });
-                }
+            CRM.confirm({title: action.title, message: msg})
+              .on('crmConfirm:yes', function () {
+                $scope.refresh([['Case', 'merge', {case_id_1: cases[0].id, case_id_2: cases[1].id}]]);
               });
-
-              var dialog = CRM.confirm({
-                title: action.title,
-                message: msg,
-                open: function () {
-                  $('input[name=change_case_status]', this).crmSelect2({data: statuses});
-                  CRM.wysiwyg.create('#change_case_status_details').then(function () {
-                    alignDialogBoxCenter(dialog);
-                  });
-                }
-              })
-                .on('crmConfirm:yes', function () {
-                  var status = $('input[name=change_case_status]', this).val();
-                  var details = $('#change_case_status_details').val();
-                  var calls = [];
-
-                  if (status) {
-                    _.each(cases, function (item) {
-                      var subject = ts('Case status changed from %1 to %2', {
-                        1: item.status,
-                        2: _.result(_.find(statuses, {id: status}), 'text')
-                      });
-                      calls.push(['Case', 'create', {id: item.id, status_id: status}]);
-                      calls.push(['Activity', 'create', {case_id: item.id, status_id: 'Completed', activity_type_id: 'Change Case Status', subject: subject, details: details}]);
-                    });
-                    $scope.refresh(calls);
-                  }
+          },
+
+          changeStatus: function (cases) {
+            var types = _.uniq(_.map(cases, 'case_type_id'));
+            var currentStatuses = _.uniq(_.collect(cases, 'status_id'));
+            var currentStatus = currentStatuses.length === 1 ? currentStatuses[0] : null;
+            var msg = '<form>' +
```

Comment:
Just wondering. can we use template string literals here
```js
 var msg = `<form>
              <div>
                <input name="change_case_status" placeholder="${ts('Select New Status')}" />
              </div>
              <label for="change_case_status_details">${ts('Notes')} </label>
              <textarea id="change_case_status_details"></textarea>
            </form>`;
```

Reply:
That is ES6 which we are not using at the moment.